code/controllers/subsystem/mapping.dm

```diff
@@ -250,17 +249,8 @@ SUBSYSTEM_DEF(mapping)
 
 ///NSV13 RECODE OF MINING LOAD SELECTION
 	//Load Mining
-	switch(config.minetype)
-		if("lavaland")
-			LoadGroup(FailedZs, "Lavaland", "map_files/Mining", "Lavaland.dmm", default_traits = ZTRAITS_LAVALAND)
-		if("nostromo") //nsv13 mining type
-			LoadGroup(FailedZs, "nostromo", "map_files/Mining/nsv13", "nostromo.dmm", default_traits = ZTRAITS_BOARADABLE_SHIP)
-		if("FOB") //nsv13 mining type
-			LoadGroup(FailedZs, "FOB", "map_files/Mining/nsv13", "FOB_Shuttle.dmm", default_traits = ZTRAITS_BOARADABLE_SHIP)
-		if(null)
-			INIT_ANNOUNCE("WARNING: A null minetype was set! Inspect the map definition JSON!")
-		else
-			INIT_ANNOUNCE("WARNING: An unknown minetype '[config.minetype]' was set! This is being ignored! Update the maploader code!")
+	instance_overmap(_path=config.mining_ship_type, folder= config.mine_path ,interior_map_files = config.mine_file, traits=config.mine_traits)
```

Comment:
This is a suitable replacement. The only issue I have with it is the _path= that is a bit hacky to manage it.